packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -967,6 +967,24 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             metadata,
         );
 
+        const shouldRunExpiry = this.runtimeOptions.gcOptions?.gcAllowed === true
+        && this.runtimeOptions.gcOptions?.disableGC !== true
+        && this.runtimeOptions.gcOptions?.runSweep === true;
+
+        if(shouldRunExpiry) {
+            const defaultContainerRuntimeExpiryMs = 30 * 24 * 60 * 60 * 1000;
+            const expiryMs = this.runtimeOptions.gcOptions?.testMode === true ?
+                0 : defaultContainerRuntimeExpiryMs;
+            const closeRuntime = () => this.closeFn({
+                errorType: "ClientSessionExpired",
+                message: `The client has reached the expiry time of ${expiryMs} ms.`,
+            });
+            const timeout = setTimeout(() => closeRuntime(), expiryMs);
```

Comment:
It may be a good idea to store this timeout in a member in the class and clear the timeout from within dispose. That way we can clear all state from a single place.